File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/ObservationEntity.java
package org.openmrs.mobile.databases.entities;

import org.openmrs.mobile.models.Resource;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "observations")
public class ObservationEntity extends Resource {

    @NonNull
    @ColumnInfo(name = "encounter_id")
    private int encounterKeyID;

    @ColumnInfo(name = "displayValue")
    private String displayValue;

    @ColumnInfo(name = "diagnosisOrder")
    private String diagnosisOrder;

    @ColumnInfo(name = "diagnosisList")
    private String diagnosisList;

    @ColumnInfo(name = "diagnosisCertainty")
    private String diagnosisCertainty;

    @ColumnInfo(name = "diagnosisNote")
    private String diagnosisNote;

    @ColumnInfo(name = "conceptUuid")
    private String conceptuuid;

    public ObservationEntity() {
    }

    public void setEncounterKeyID(int encounterKeyID) {
        this.encounterKeyID = encounterKeyID;
    }

    public void setDisplayValue(String displayValue) {
        this.displayValue = displayValue;
    }

    public void setDiagnosisOrder(String diagnosisOrder) {
        this.diagnosisOrder = diagnosisOrder;
    }

    public void setDiagnosisList(String diagnosisList) {
        this.diagnosisList = diagnosisList;
    }

    public void setDiagnosisCertainty(String diagnosisCertainty) {
        this.diagnosisCertainty = diagnosisCertainty;
    }

    public void setDiagnosisNote(String diagnosisNote) {
        this.diagnosisNote = diagnosisNote;
    }

    public void setConceptuuid(String conceptuuid) {
        this.conceptuuid = conceptuuid;
    }

    public int getEncounterKeyID() {
        return encounterKeyID;
    }

    public String getDisplayValue() {
        return displayValue;
    }

    public String getDiagnosisOrder() {
        return diagnosisOrder;
    }

    public String getDiagnosisList() {
        return diagnosisList;
    }

    public String getDiagnosisCertainty() {
        return diagnosisCertainty;
    }

    public String getDiagnosisNote() {
        return diagnosisNote;
    }

    public String getConceptuuid() {
        return conceptuuid;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ObservationEntity)) return false;

        ObservationEntity entity = (ObservationEntity) o;

        if (encounterKeyID != entity.encounterKeyID) return false;
        if (displayValue != null ? !displayValue.equals(entity.displayValue) : entity.displayValue != null)
            return false;
        if (diagnosisOrder != null ? !diagnosisOrder.equals(entity.diagnosisOrder) : entity.diagnosisOrder != null)
            return false;
        if (diagnosisList != null ? !diagnosisList.equals(entity.diagnosisList) : entity.diagnosisList != null)
            return false;
        if (diagnosisCertainty != null ? !diagnosisCertainty.equals(entity.diagnosisCertainty) : entity.diagnosisCertainty != null)
            return false;
        if (diagnosisNote != null ? !diagnosisNote.equals(entity.diagnosisNote) : entity.diagnosisNote != null)
            return false;
        return conceptuuid != null ? conceptuuid.equals(entity.conceptuuid) : entity.conceptuuid == null;
    }

    @Override
    public int hashCode() {
        int result = encounterKeyID;
        result = 31 * result + (displayValue != null ? displayValue.hashCode() : 0);
        result = 31 * result + (diagnosisOrder != null ? diagnosisOrder.hashCode() : 0);
        result = 31 * result + (diagnosisList != null ? diagnosisList.hashCode() : 0);
        result = 31 * result + (diagnosisCertainty != null ? diagnosisCertainty.hashCode() : 0);
        result = 31 * result + (diagnosisNote != null ? diagnosisNote.hashCode() : 0);
        result = 31 * result + (conceptuuid != null ? conceptuuid.hashCode() : 0);
        return result;
    }
}

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/ObservationEntity.kt
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package org.openmrs.mobile.databases.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import org.openmrs.mobile.models.Resource

@Entity(tableName = "observations")
class ObservationEntity : Resource() {
    @ColumnInfo(name = "encounter_id")
    @SerializedName("encounter_id")
    @Expose
    var encounterKeyID: Long = 0

    @ColumnInfo(name = "displayValue")
    @SerializedName("displayValue")
    @Expose
    var displayValue: String? = null

    @ColumnInfo(name = "diagnosisOrder")
    @SerializedName("diagnosisOrder")
    @Expose
    var diagnosisOrder: String? = null

    @ColumnInfo(name = "diagnosisList")
    @SerializedName("diagnosisList")
    @Expose
    var diagnosisList: String? = null

    @ColumnInfo(name = "diagnosisCertainty")
    @SerializedName("diagnosisCertainty")
    @Expose
    var diagnosisCertainty: String? = null

    @ColumnInfo(name = "diagnosisNote")
    @SerializedName("diagnosisNote")
    @Expose
    var diagnosisNote: String? = null

    @ColumnInfo(name = "conceptUuid")
    @SerializedName("conceptUuid")
    @Expose
    var conceptuuid: String? = null

}
